README.md

```diff
@@ -134,6 +134,50 @@ until the constructor for `config.Provider` can be fully satisfied.
 If resolution is not possible, for instance one of the required dependencies has
 does not have a constructor and doesn't appear in the graph, an error will be returned.
 
+## Invoke
+
+`Invoke` API executes the provided function and stores return results in the dig container.
+
+`Invoke` looks through the dig graph and resolves all the constructor parameters for execution.
+The return results, except the `error` object is inserted into the graph for further use.
+To utilize `Invoke`, function needs following form -
+
+1. function arguments are pointers, maps, slice or arrays
+2. function arguments must be provided to dig
+3. function return parameters are optional
+4. provided function can be anonymous
+
+For example, `Invoke` scenarios:
+
+Invoke anonymous function:
+```go
+ // c := dig.New()
+// c.Provide... config.Provider, zap.Logger, etc
+err := c.Invoke(func(cfg *config.Provider, l *zap.Logger) error {
+  // function body
+	return nil
+})
+```
+
+Invoke recursively resolve dependencies:
```

Comment:
This sentence doesn't seem quite right. Do you mean "recursively resolved dependencies"?